src/lib/kademlia/membership.ml

```diff
@@ -54,6 +54,43 @@ let get_project_root () =
   in
   go @@ realpath current_dir_name
 
+(* This snippet was taken from our fork of RPC Parallel.
+ * Would be nice to have a shared utility, but this is
+ * easiest for now. *)
+(* To get the currently running executable:
+  On Darwin:
+  Use _NSGetExecutablePath via Ctypes
+
+  On Linux:
+  Use /proc/PID/exe
+   - argv[0] might have been deleted (this is quite common with jenga)
+   - `cp /proc/PID/exe dst` works as expected while `cp /proc/self/exe dst` does not *)
+let get_coda_binary =
+  let our_binary_lazy =
+    lazy
+      (let open Async in
+      let open Deferred.Or_error.Let_syntax in
+      let%map os = Process.run ~prog:"uname" ~args:["-s"] () in
+      if os = "Darwin\n" then (
+        let open Ctypes in
+        let ns_get_executable_path =
+          Foreign.foreign "_NSGetExecutablePath"
+            (ptr char @-> ptr uint32_t @-> returning void)
+        in
+        let path_max = 1024 in
```

Comment:
is it possible to pull in `sys/syslimits.h` with Ctypes? Hardcoding the constant makes me a bit nervous.

Reply:
Yes, it's possible. I'll just wrap a function around the preprocessor value.